src/content/app/genome-browser/components/browser-track-config/useBrowserTrackConfig.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useRef } from 'react';
+import { RootState } from 'src/store';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { OptionValue } from 'src/shared/components/radio-group/RadioGroup';
+
+import { getBrowserActiveGenomeId } from '../../state/browser-general/browserGeneralSelectors';
+import {
+  getTrackConfigForTrackId,
+  getApplyToAllConfig,
+  getBrowserCogList,
+  getBrowserSelectedCog
+} from '../../state/track-config/trackConfigSelectors';
+import {
+  updateTrackConfigNames,
+  updateTrackConfigLabel,
+  updateApplyToAll,
+  TrackType
+} from '../../state/track-config/trackConfigSlice';
+
+import analyticsTracking from 'src/services/analytics-service';
+import useGenomeBrowser from '../../hooks/useGenomeBrowser';
+
+const useBrowserTrackConfig = () => {
+  const applyToAllConfig = useSelector(getApplyToAllConfig);
+  const browserCogList = useSelector(getBrowserCogList);
+  const selectedCog = useSelector(getBrowserSelectedCog) || '';
+  const activeGenomeId = useSelector(getBrowserActiveGenomeId);
+  const shouldApplyToAll = applyToAllConfig.isSelected;
+  const shouldApplyToAllRef = useRef(shouldApplyToAll);
+
+  const selectedTrackConfigInfo = useSelector((state: RootState) =>
+    getTrackConfigForTrackId(state, selectedCog)
+  );
+
+  useEffect(() => {
+    shouldApplyToAllRef.current = shouldApplyToAll;
+  }, [shouldApplyToAll]);
+
+  const dispatch = useDispatch();
+
+  const { toggleTrackName, toggleTrackLabel } = useGenomeBrowser();
+
+  const updateTrackName = (isTrackNameShown: boolean) => {
+    if (!activeGenomeId) {
+      return;
+    }
+    if (shouldApplyToAllRef.current) {
```

Comment:
Is this not the decision that should be made in the reducer? The reducer should know whether the "shouldApplyToAll" is selected, and would apply the new setting to all relevant tracks, I would have thought.

By the way, is the "apply to all" state shared between all configs? I.e. if I switch the radio button to "apply to all" in one track config panel, and then open another track config panel, does the radio button there show as selected?

Reply:
yes, "apply to all" is global as far as I know